node/pkg/vaa/structs_test.go

```diff
@@ -0,0 +1,117 @@
+package vaa
+
+import (
+	"encoding/hex"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestChainIDFromString(t *testing.T) {
+	type test struct {
+		i string
+		o ChainID
+	}
+
+	// Positive Test Cases
+	p_tests := []test{
+		{i: "solana", o: ChainIDSolana},
+		{i: "ethereum", o: ChainIDEthereum},
+		{i: "terra", o: ChainIDTerra},
+		{i: "bsc", o: ChainIDBSC},
+		{i: "polygon", o: ChainIDPolygon},
+		{i: "avalanche", o: ChainIDAvalanche},
+		{i: "oasis", o: ChainIDOasis},
+		{i: "fantom", o: ChainIDFantom},
+		{i: "algorand", o: ChainIDAlgorand},
+		{i: "ethereum-ropsten", o: ChainIDEthereumRopsten},
+		{i: "Solana", o: ChainIDSolana},
+		{i: "Ethereum", o: ChainIDEthereum},
+		{i: "Terra", o: ChainIDTerra},
+		{i: "Bsc", o: ChainIDBSC},
+		{i: "Polygon", o: ChainIDPolygon},
+		{i: "Avalanche", o: ChainIDAvalanche},
+		{i: "Oasis", o: ChainIDOasis},
+		{i: "Fantom", o: ChainIDFantom},
+		{i: "Algorand", o: ChainIDAlgorand},
+		{i: "Karura", o: ChainIDKarura},
+		{i: "Acala", o: ChainIDAcala},
+	}
+
+	// Negative Test Cases
+	n_tests := []test{
+		{i: "Unknown", o: ChainIDUnset},
+	}
+
+	for _, tc := range p_tests {
+		chainId, err := ChainIDFromString(tc.i)
```

Comment:
```suggestion
	for _, tc := range p_tests {
		t.Run(tc.i, func(t *testing.T) {
	        	chainId, err := ChainIDFromString(tc.i)
```

Make these all individual test cases.